src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyMetaPackages.cs

```diff
@@ -56,13 +57,35 @@ public override bool Execute()
             var index = PackageIndex.Load(PackageIndexes.Select(pi => pi.GetMetadata("FullPath")));
             List<ITaskItem> updatedDependencies = new List<ITaskItem>();
 
-            var suppressMetaPackages = new HashSet<string>(SuppressMetaPackages.NullAsEmpty(), StringComparer.OrdinalIgnoreCase);
+            var suppressMetaPackages = new Dictionary<string, HashSet<string>>();
+
+            if (SuppressMetaPackages != null)
+            {
+                foreach (var metapackage in SuppressMetaPackages)
+                {
+                    if (!suppressMetaPackages.ContainsKey(metapackage.ItemSpec))
+                    {
+                        suppressMetaPackages.Add(metapackage.ItemSpec, new HashSet<string>());
+                    }
+                    var tfmSpecificSupression = metapackage.GetMetadata("TargetFramework");
+                    if (string.IsNullOrEmpty(tfmSpecificSupression))
+                    {
+                        // If the supression doesn't specify a TargetFramework, then it applies to all.
+                        suppressMetaPackages[metapackage.ItemSpec].Add("All");
+                    }
+                    else
+                    {
+                        var fx = NuGetFramework.Parse(tfmSpecificSupression);
+                        suppressMetaPackages[metapackage.ItemSpec].Add(fx.DotNetFrameworkName);
+                    }
```

Comment:
We could simplify this:

```suggestion
                    if (!suppressMetaPackages.TryGetValue(metapackage.ItemSpec, out var value))
                    {
                        value = new HashSet<string>();
                        suppressMetaPackages.Add(metapackage.ItemSpec, value);
                    }
                    var tfmSpecificSupression = metapackage.GetMetadata("TargetFramework");
                    if (string.IsNullOrEmpty(tfmSpecificSupression))
                    {
                        // If the supression doesn't specify a TargetFramework, then it applies to all.
                        value.Add("All");
                    }
                    else
                    {
                        var fx = NuGetFramework.Parse(tfmSpecificSupression);
                        value.Add(fx.DotNetFrameworkName);
                    }
```